src/payment/strategies/afterpay/afterpay-payment-strategy.ts

```diff
@@ -68,11 +73,17 @@ export default class AfterpayPaymentStrategy implements PaymentStrategy {
 
         await this._checkoutValidator.validate(state.checkout.getCheckout(), options);
 
-        state = await this._store.dispatch(
-            this._paymentMethodActionCreator.loadPaymentMethod(paymentId, options)
-        );
+        state = await this._loadPaymentMethod(gatewayId, methodId, options);
+
+        const checkout = state.checkout.getCheckout();
+        const amount = checkout?.outstandingBalance || 0;
+        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId, gatewayId);
+
+        if (this._outsideLimits(amount, paymentMethod)) {
```

Comment:
If the amount was outside of the limits, would we not have thrown the error at line `76` already?

Reply:
@leeBigCommerce This case covers an issue with the store credit, they are initializing in a different way, that's why we went with this double validation